fix: split text by any newline and spaces

franluca opened the pull request:
*Description of changes:* when computing f1 measures for QA, split the text by any number of newlines or spaces (previously was considering only one new line). 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

bilalaws commented on src/fmeval/eval_algorithms/qa_accuracy.py:
```diff
@@ -124,7 +134,7 @@ def _f1_score(
     model_output = _normalize_and_strip_text(model_output, normalize_text=normalize_text, strip_text=strip_text)
     target_output = _normalize_and_strip_text(target_output, normalize_text=normalize_text, strip_text=strip_text)
-    ret = f_measure(reference=set(target_output.split(" ")), test=set(model_output.split(" ")))
+    ret = f_measure(reference=set(_split(target_output)), test=set(_split(model_output)))
```
Do we really need a separate function? Could we not replace `split(" ")` with `split()`. Simply using `split()` gives the desired behavior that you specified in the unit tests.

franluca replied:
Ah makes sense. I didn't know the behavior of split with no separator. Actually from the docs the behavior is not that clear. https://docs.python.org/3.10/library/stdtypes.html?highlight=str%20split#str.split  
there is no mention about newlines and other special characters. But yes str.split() seems to achieve precisely what we need. Will change.

bilalaws replied:
The only thing `_split` does is to call the standard library `split()` function, so consider calling `split()` directly. Its just one more level that the reader has to go through and not sure that it adds any clarity since the user can simply look at the documentation of standard library `split()` to understand what it does.

bilalaws commented on src/fmeval/eval_algorithms/qa_accuracy.py:
```diff
@@ -105,6 +106,15 @@ def _normalize_and_strip_text(text: str, *, normalize_text: bool = False, strip_
+    """
+    Split text matching one or more spaces (\s+) or one or more new lines (\n+).
+    E.g. re.split(r"\n+\s+", 'True\n\nI    disagree') = ['True', 'I', 'disagree'].
+    Note the
```
incomplete line

franluca commented:
Made changes to documentation, added a test case and corrected lint. 

I think it's fine to have one separate function for split even if it calls only str.split . This adds clarity to the code.

bilalaws commented on src/fmeval/eval_algorithms/qa_accuracy.py:
```diff
@@ -105,6 +105,17 @@ def _normalize_and_strip_text(text: str, *, normalize_text: bool = False, strip_
 
+def _split(text: str) -> List[str]:
+    """
+    Split text matching one or more spaces (\s+) or one or more new lines (\n+) or any other string.whitespace
```
For the sake of conciseness, consider updating this to: Split text based on `string.whitespace` characters.
Reason: space and newline are included in `string.whitespace` characters.

bilalaws reviewed (dismissed):
Thanks Luca. Left a couple of non-blocking comments.
